internal/page/donor/upload_evidence.go

```diff
@@ -74,46 +83,81 @@ func UploadEvidence(tmpl template.Template, payer Payer, donorStore DonorStore,
 			if data.Errors.None() {
 				switch form.Action {
 				case "upload":
+					var uploadedDocuments []page.Document
+
 					for _, file := range form.Files {
-						uuid := randomUUID()
-						key := lpa.UID + "/evidence/" + uuid
+						key := lpa.UID + "/evidence/" + randomUUID()
 
-						err := evidenceS3Client.PutObject(r.Context(), key, file.Data)
-						if err != nil {
+						document := page.Document{
+							PK:       "LPA#" + lpa.ID,
+							SK:       "#DOCUMENT#" + key,
+							Filename: file.Filename,
+							Key:      key,
+						}
+						if err := documentStore.Put(r.Context(), document, file.Data); err != nil {
```

Comment:
it would be nice to provide a `documentStore.Create(ctx, lpa.ID, key, file.Filename)` method, then you won't have `LPA#` and `#DOCUMENT#` exposed here